app/mapping/registration/IndividualBeneficiaryMapper.scala

```diff
@@ -50,6 +50,25 @@ class IndividualBeneficiaryMapper @Inject()(addressMapper: AddressMapper) extend
     }
   }
 
+  def removeIncompleteBeneficiaries(beneficiaries: List[IndividualBeneficiary], userAnswers: UserAnswers): List[IndividualBeneficiary] = {
+    beneficiaries.filter(indDetailsType =>
+      (userAnswers.is5mldEnabled, userAnswers.isTaxable) match {
+        case (true, true) =>
+            indDetailsType.incomeYesNo.isDefined &&
+            indDetailsType.vulnerableYesNo.isDefined &&
+            indDetailsType.countryOfResidence.isDefined &&
+            indDetailsType.countryOfNationality.isDefined &&
```

Comment:
I'm not sure I agree with this. Country of residence and country of nationality can be None if the user answers no to the "do you know the country of nationality/residence?" question

Reply:
Yeah, your correct